### PR TITLE
Bug fix initialization of cross-validation folds

### PR DESCRIPTION
**Pull request by mpecha**
This pull request contains bug fix of initializationing cross-validation folds and example 2.

### CODE REVIEW

File: examples/ex2.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/*TEST
+
+  test:
+    filter: grep -v MPI
+    args: -f_training $PERMON_SVM_DIR/examples/data/heart_scale.bin -f_test $PERMON_SVM_DIR/examples/data/heart_scale.t.bin
+    args: -cross_svm_view -cross_svm_view_score -cross_qps_view_convergence
+    args: -svm_view -svm_view_score -qps_view_convergence
+    output_file: output/ex2.out
+TEST*/
```
Comment (jkruzik) [inline examples/ex2.c]: new line

File: examples/output/ex2.out [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+    number of proportioning steps 6
+  type: binary
+  model parameters:
+    ||w||=3.2700    bias=0.0000    margin=0.6116    NSV=30
+  L1 hinge loss:
+    sum(xi_i)=46.5031
+  objective functions:
+    primalObj=730.2585    dualObj=701.9622    gap=28.2964
+  type: binary
+  model performance score with training parameters C=15.588, mod=2, loss=L1:
```
Comment (jkruzik) [inline examples/output/ex2.out]: C=3^(5/2) while its specified that max C = 3^2. The bug is in [SVMInitGridSearch_Binary_Private:1634](https://github.com/permon/permonsvm/blob/97a5403338c84b04a763c0859db04815d042120c/src/svm/impls/binary/binary.c#L1634) as (end- start +1)/step = 10 maybe (end- start +step)/step = 9 would work. It does also work e.g., for start=0, end=2, step=2 where the original formula gives 1 and mine gives 2. But it should be verified that it always works.

File: examples/output/ex2.out [hunk 3]
```diff
@@ -0,0 +1,744 @@
...
+    Confusion matrix:
+      TP =   20      FP =    3
+      FN =    4      TN =   33
+    accuracy=88.33%    precision=86.96%    sensitivity=83.33%
+    F1=0.85    MCC=0.76    AUC_ROC=0.88    G1=0.75
+  type: mpgp
+  last QPSSolve CONVERGED due to CONVERGED_RTOL, KSPReason=2, required 46 iterations
+  all 1 QPSSolves from last QPSReset/QPSResetStatistics have required 46 iterations
```
Comment (jkruzik) [inline examples/output/ex2.out:651]: what is this solve?

**Review by jkruzik — CHANGES_REQUESTED** [review]

File: examples/output/ex2.out (hunk 3 above)
Comment (mpecha) [reply examples/output/ex2.out:651]: I do not understand the question. Please, ask more precisely.

File: examples/output/ex2.out (hunk 2 above)
Comment (mpecha) [reply examples/output/ex2.out]: You're right, thanks. An original formula works only for step-size equal 1. Verification of your formula is simple: (end - start) / step gives you a number of variables between 'start' and 'end' (end included) by discretization step and we o want to include start - therefore we add 1 (step / step in your formula). Fixed.

File: examples/output/ex2.out (hunk 3 above)
Comment (jkruzik) [reply examples/output/ex2.out:651]: Resolved by personal communication - the solve is for training of the whole dataset

**Review by jkruzik — APPROVED** [review]

**Pull request merged**